src/core/namespace.c

```diff
@@ -2461,7 +2461,7 @@ int setup_namespace(
                 r = mount_nofollow_verbose(LOG_DEBUG, root, root, NULL, MS_BIND|MS_REC, NULL);
                 if (r < 0)
                         goto finish;
-                r = mount_move_root(root);
+                r = mount_pivot_root(root);
```

Comment:
hmm, what about mount_move_root() use in nspawn? shouldn't we switch that over too and then kill mount_move_root() for good?

Reply:
ah, i see you answered this already, so ignore me

Reply:
It's a bit more complicated from looking at it because nspawn mucks around with MS_SHARED which is something that pivot_root() isn't found of for both the old and new root. I need to take a closer look at this before we can convert this.